### DISCUSSION TITLE
large memory leak with CoPilot vscode on Fedora 39

### DISCUSSION THREAD

**Discussion by tjdeveng** [community/community#102205] [Copilot Conversations]
### Select Topic Area

Bug

### Body

Have been trying copilot extensions for a few days now and have noticed a memory leak causing sluggish response from VS code and code is shut down after approximately half hour or so due to its consuming most of the memory on my system.
Have tested my stopping all other extensions and monitoring the resources using top, only when copilot is enabled does the memory usage start to escalate (approx 10Mb every five seconds). I am running under a VM with 16Gb RAM.
I have found a similar issue but the conversation was in 2022, so I believe this may be a new issue/regression.

**Comment by Akash1134** [comment]
Hi @tjdeveng ,

Please send us the details of your local VSCode Copilot setup. 

- Navigate to View > Command Palette.. and type `GitHub Copilot Collect Diagnostics`
- You'll  be shown a file that includes shareable Network, Extension and Environment details.

Thanks!

**Reply by tjdeveng** [reply]
Thank you for your response,

The info requested is:
## Copilot

- Version: 1.156.0
- Build: prod
- Editor: vscode/1.85.2

## Environment

- http_proxy: n/a
- https_proxy: n/a
- no_proxy: n/a
- SSL_CERT_FILE: n/a
- SSL_CERT_DIR: n/a
- OPENSSL_CONF: n/a

## Feature Flags

- Custom Certificates: disabled
- Send Restricted Telemetry: disabled
- Chat: enabled

## Node setup

- Number of root certificates: 140
- Operating system: Linux
- Operating system version: 6.6.13-200.fc39.x86_64
- Operating system architecture: x64
- NODE_OPTIONS: n/a
- NODE_EXTRA_CA_CERTS: n/a
- NODE_TLS_REJECT_UNAUTHORIZED: n/a
- tls default min version: TLSv1.2
- tls default max version: TLSv1.3

## Network Configuration

- Proxy host: n/a
- Proxy port: n/a
- Kerberos SPN: n/a
- Reject unauthorized: disabled

## Reachability

- github.com: HTTP 200 - 
- copilot-proxy.githubusercontent.com: HTTP 200 - 
- api.githubcopilot.com: HTTP 200 - 
- default.exp-tas.com: HTTP 200 - 

## VS Code Configuration

- HTTP proxy: 
- HTTP proxy autentication: n/a
- Proxy Strict SSL: true
- Extension HTTP proxy support: override

## Extensions

- Is `win-ca` installed?: false
- Is `mac-ca` installed?: false
-
